irohad/ametsuchi/impl/postgres_block_index.cpp

```diff
@@ -76,27 +76,32 @@ namespace iroha {
                   "index) "
                   "VALUES ("
                   + transaction_.quote(id) + ", " + transaction_.quote(height)
-                  + ", " + transaction_.quote(cmd->asset_id) + ", "
+                  + ", " + transaction_.quote(command->assetId()) + ", "
                   + transaction_.quote(index) + ");");
               status &= res != boost::none;
             });
             return status;
           });
     }
 
-    void PostgresBlockIndex::index(const model::Block &block) {
-      const auto &height = std::to_string(block.height);
+    void PostgresBlockIndex::index(
+        const w<shared_model::interface::Block> block) {
+      const auto &height = std::to_string(block->height());
       boost::for_each(
-          block.transactions | boost::adaptors::indexed(0),
+          block->transactions() | boost::adaptors::indexed(0),
           [&](const auto &tx) {
-            const auto &creator_id = tx.value().creator_account_id;
-            const auto &hash = iroha::hash(tx.value()).to_string();
+            const auto &creator_id = tx.value()->creatorAccountId();
+            const auto &hash = tx.value()->hash();
             const auto &index = std::to_string(tx.index());
 
+            std::string tmp;
+            for (const auto c : hash.blob()) {
+              tmp += c;
+            }
```

Comment:
looks very weird. Is it a some kind of a cast from blob to string?

if so, why don't u just write 
```
auto blob = hash.blob();
std::string tmp{blob.begin(), blob.end()};
```

Reply:
Used that as temporary solution, then forgot to make it pretty. 
Will change it